msal/src/main/java/com/microsoft/identity/client/AccountAdapter.java

```diff
@@ -60,7 +60,9 @@ public List<ICacheRecord> filter(@NonNull List<ICacheRecord> records) {
                 final String acctHomeAccountId = cacheRecord.getAccount().getHomeAccountId();
                 final String acctLocalAccountId = cacheRecord.getAccount().getLocalAccountId();
 
-                if (!acctHomeAccountId.contains(acctLocalAccountId)) {
+                if (acctHomeAccountId != null
+                        && acctLocalAccountId != null
+                        && !acctHomeAccountId.contains(acctLocalAccountId)) {
                     result.add(cacheRecord);
```

Comment:
You may also want to add a warning log to say that these values are null. Ideally, they should not be null.